pex/pex_info.py

```diff
@@ -274,11 +289,13 @@ def update(self, other):
       raise TypeError('Cannot merge a %r with PexInfo' % type(other))
     self._pex_info.update(other._pex_info)
     self._distributions.update(other.distributions)
+    self._interpreter_constraints.extend(other.interpreter_constraints)
```

Comment:
this should probably dedupe somehow, perhaps via internal storage as a `set`?

Reply:
Python sets are not JSON serializable but I can use sets to reduce duplications and cast them as lists.

Reply:
sounds good